### DISCUSSION TITLE
How to define the root folder for my qmd content

### DISCUSSION THREAD

**Discussion by dweng0** [quarto-dev/quarto-cli#6093] [Q&A] [answered]
### Description

Is there a way to explicitly tell quarto where root content folder is? By root content folder I mean all the content that is to be published. I have taken a look at the [_quarto.yml](https://github.com/quarto-dev/quarto-web/blob/main/_quarto.yml) and can't see any thing that dictates the root file. I also searched through the QA discussion and couldn't find anything. Is it possible?

### My problem
My root content is held in a folder called `content`. When quarto renders this, the url  shows as `localhost/content`. But really i want it to just be `localhost`. Ie content folder should be treated as the root.

**Comment by mcanouil** [comment, marked as answer]
At first I thought you were looking for `working-dir`, see <https://quarto.org/docs/projects/code-execution.html#working-dir>.

But it's unclear what you want to have.
Quarto will keep your source code directory tree structure, because it has no reason to flatten this.
You can use aliases or as you seem to do `_redirects` Netlify configuration file to make it look like the files are served from your website root, see <https://quarto.org/docs/websites/website-navigation.html#redirects>.

---

For a more precise answer, you might want to consider sharing a small self-contained "working" (reproducible) example to work with, _i.e._, a complete Quarto document or a Git repository.

You can share a Quarto document using the following syntax, _i.e._, using more backticks than you have in your document (usually four ` ```` `).

`````md
````qmd
---
title: "Reproducible Quarto Document"
format: html
---

This is a reproducible Quarto document using `format: html`.
It is written in Markdown and contains embedded R code.
When you run the code, it will produce a plot.

```{r}
plot(cars)
```

The end.
````
`````

**Reply by dweng0** [reply]
As i was writing this QA I was wondering if my directory stratergy was correct for Quarto, your comment has confirmed it for me. will adjust my directory structure

**Comment by allefeld** [comment]
I have the same question. I'm working on software in a GitHub repository `project`, and I want to publish its documentation via GitHub pages from the same repository. To keep things orderly, I have the documentation source files in a separate subdirectory `/docs-sources`. The documentation pages should be in the root of the served web page, e.g. `https://allefeld.github.io/project/index.html`.

When I render the source files through with this configuration in `_quarto.yml`:

```yaml
project:
  type: website
  render:
    - docs-sources/
  output-dir: docs/
```

the rendered files get put into `/docs/doc-sources/`, where the subdirectory is unnecessary:

```
project/
    _quarto.yml
    docs-sources/
        index.qmd
    docs/
        index.html        <- contains only redirect
        docs-sources/
            index.html     <- rendered from index.qmd
```

And if I select `/docs` as the source for GitHub pages, visiting `https://allefeld.github.io/project/` redirects me to `https://allefeld.github.io/cvcrossmanova/docs-sources/index.html`, i.e. what is just an organizational detail of my repository shows up in the URL.

What I'm looking for is to have the rendering ignore the `\docs-sources` subdirectory, take the files from there as if they were in `/`, and put the output files directly into `\docs`.

```
project/
    _quarto.yml
    docs-sources/
        index.qmd
    docs/
        index.html     <- rendered from index.qmd
```

I think 'root folder' is a reasonable description for this. E.g.

```yaml
project:
  type: website
  render-root: docs-sources/
  output-dir: docs/
```

Previously I tried to put the Quarto project root (and therefore `_quarto.yml`) into the `\docs-sources`  subdirectory. But then ~~I can't render into a directory one level up, and~~ VS Code gets confused when I try to render the Quarto Project while having the repository root open.

**Reply by allefeld** [reply]
Okay, seems I was wrong about not being able to render one level up. A workaround is to use the setup mentioned in the last paragraph, with

```yaml
project:
  type: website
  output-dir: ../docs/
```

and to render the project only from outside VS Code, from the command line.

Interestingly, I can render individual files using the standard shortcut, and `_quarto.yml` settings are respected.

Seems like I found my solution, but maybe it would be worth mentioning this setup somewhere in the documentation.
